test/api.test.js

```diff
@@ -535,6 +545,74 @@ describe("api.js", () => {
             }]);
         });
 
+        it("should fail by timeout during rendition processing", async () => {
+            delete process.env.DISABLE_IO_EVENTS_ON_TIMEOUT;
+
+            const receivedMetrics = MetricsTestHelper.mockNewRelic();
+            testUtil.nockIOEvent({
+                type: "rendition_failed",
+                errorReason: "GenericError",
+                errorMessage: 'Timeout',
+                rendition: {
+                    fmt: "png"
+                },
+                source: "https://example.com/MySourceFile.jpg"
+            });
+
+            const main = worker(async function(source, rendition) {
+                fs.writeFileSync(rendition.path, testUtil.RENDITION_CONTENT);
+                await sleep(500);
+                console.log('waiting...');
+                return Promise.resolve();
+            });
+            assert.equal(typeof main, "function");
+            process.env.__OW_DEADLINE = Date.now() + 300;
+
+            await main(testUtil.simpleParams({noEventsNock:true}));
+            testUtil.assertNockDone();
+            await MetricsTestHelper.metricsDone();
+            MetricsTestHelper.assertArrayContains(receivedMetrics, [{
+                eventType: 'timeout'
+            },{
+                eventType: "activation"
+            }]);
+            assert.equal(processSpy.calledOnce, true, "did not call process.exit(124) on timeout");
+        });
+
+        it("should fail by timeout during second rendition processing", async () => {
+            delete process.env.DISABLE_IO_EVENTS_ON_TIMEOUT;
+            const receivedMetrics = MetricsTestHelper.mockNewRelic();
+            testUtil.nockIOEvent({
+                type: "rendition_failed",
+                errorReason: "GenericError",
+                errorMessage: 'Timeout',
+                rendition: { fmt: 'xml', name: 'MyRendition3.xml' },
+                source: "https://example.com/MySourceFile.jpg"
+            });
+
+            const main = worker(async function(source, rendition) {
+                if (rendition.index === 2) {
+                    console.log('waiting...');
+                    await sleep(5000);
+                }
+                fs.writeFileSync(rendition.path, testUtil.RENDITION_CONTENT);
+                return Promise.resolve();
+            });
+            assert.equal(typeof main, "function");
+            process.env.__OW_DEADLINE = Date.now() + 7000;
+
+            await main(testUtil.paramsWithMultipleRenditions({noPut3:true}));
+
+            testUtil.assertNockDone();
+            await MetricsTestHelper.metricsDone();
+            MetricsTestHelper.assertArrayContains(receivedMetrics, [{
+                eventType: 'timeout'
+            },{
+                eventType: "activation"
+            }]);
+            assert.equal(processSpy.calledOnce, true, "did not call process.exit(124) on timeout");
+        }).timeout(10000);
```

Comment:
Also, if you reduce the wait time, maybe you can (at least) reduce the timeout value here, or even remove it.